overlord/fdestate/fdemgr.go

```diff
@@ -412,6 +413,41 @@ func GetRecoveryKey(st *state.State, keyID string) (rkey keys.RecoveryKey, err e
 	return mgr.getRecoveryKey(keyID)
 }
 
+// CheckRecoveryKey tests that the specified recovery key unlocks the
+// specified container roles. If no container roles are passed, the recovery
+// key will be tested against all container roles.
+//
+// The state must be locked by the caller.
+func (m *FDEManager) CheckRecoveryKey(rkey keys.RecoveryKey, containerRoles []string) error {
+	containers, err := m.GetEncryptedContainers()
+	if err != nil {
+		return err
+	}
+
+	allContainers := len(containerRoles) == 0
+
+	devPathByContainerRole := make(map[string]string, len(containers))
+	for _, container := range containers {
+		devPathByContainerRole[container.ContainerRole()] = container.DevPath()
+		if allContainers {
+			containerRoles = append(containerRoles, container.ContainerRole())
+		}
+	}
```

Comment:
will we need to do this in other places?  maybe at least a TODO about creating a helper?

Reply:
The only place I think this would have been reused is the GetKeyslots helper, and now after trying to use the helper it feels a little forced and unnecessary, In the latest changed I moved away from `devPathByContainerRole` and refactored the code there where it is not needed.